web/modules/custom/joinup_news/src/Guard/FulfillmentGuard.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace Drupal\joinup_news\Guard;
+
+use Drupal\og\Og;
+use Drupal\state_machine\Guard\GuardInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowTransition;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Guard class for the transitions of the news entity.
+ *
+ * @package Drupal\joinup_news\Guard
+ */
+class FulfillmentGuard implements GuardInterface {
+  /**
+   * Constant representing the pre-moderating parent.
+   */
+  const MODERATED = 1;
+
+  /**
+   * Constant representing the post-moderating parent.
+   */
+  const NON_MODERATED = 0;
+
+  /**
+   * {@inheritdoc}
+   *
+   * We need to override default transitions allowed because this is also
+   * dependant on the parent's moderation, system roles and organic groups
+   * user roles. In the following method, the allowed transitions per
+   * moderation are checked and then if the transition is allowed, the user
+   * roles by the system and the organic groups are checked.
+   */
+  public function allowed(WorkflowTransition $transition, WorkflowInterface $workflow, EntityInterface $entity) {
```

Comment:
The guard is bound to a workflow group. It is declared in the services yml.
We are going to provide one workflow for each case we have so we do not need to check if it applies as this is called only for the entities that have field of type state using this workflow group.

The connection between the workflow and the entity is:
A workflow group is defined in the workflow_group file.
A gourp (or more) is defined in the workflow file.
A service for the guard is defined marked with the tag of the workflow group id.
A state field is created and in the settings, one of the groups are selected.

So as long as we do not use this workflow for another bundle or entity type, there is no need to check.